### ISSUE TITLE
Implement backpropagation

### ISSUE THREAD

**Issue by RenaudWasTaken** [LinkedNeurons/NeuronsJustGotReal#8]
In order to implement a backpropagation algorithm you will need to complete issue #2.

After that implement a simple backpropagation algorithm (gradient descent)
